### DISCUSSION TITLE
Антизапрет и Амнезия

### DISCUSSION THREAD

**Discussion by Ernieleo** [xtrime-ru/antizapret-vpn-docker#69] [Q&A] [answered]
Всем привет. Если правильно понимаю, удалось успешно реализовать работу сервиса через модернизированный WireGuard - AmneziaWG, если всё так, хотелось бы узнать как установить работу через данный протокол и при этом сохранить возможность корректных дальнейших обновлений.

Изучив последние изменения и документацию проекта, не понял как задействовать амнезию при новой установке.
Буду благодарен за удобную в восприятии инструкцию.

**Comment by Dimariqe** [comment, marked as answer]
Можете воспользоваться [инструкцией для Keenetic](https://github.com/xtrime-ru/antizapret-vpn-docker/blob/master/docs/guide_Keenetic_RU.md#wireguard-%D1%81%D0%B5%D1%80%D0%B2%D0%B5%D1%80%D0%BD%D0%B0%D1%8F-%D1%87%D0%B0%D1%81%D1%82%D1%8C), за тем исключением, что после пункта 2 создайте пароль для панели wireguard:
```shell
docker run --rm ghcr.io/wg-easy/wg-easy wgpw 'ВАШ_ПАРОЛЬ' | sed "s/'//g" | sed -r 's/\$/\$\$/g' | tee ./wireguard/wireguard.env
```
а в файле `docker-compose.override.yml` не используйте строки:
```yaml
      # Сгенерируйте собственный пароль. Установлен пароль: TestGenPa123
      # docker run -it ghcr.io/w0rng/amnezia-wg-easy wgpw "TestGenPa123"
      # Или: https://bcrypt.ninja/
      # Заменяем все $ на $$
      - PASSWORD_HASH=$$2a$$12$$J0lBSna8bUdYhx90SjOqMOtu8O.s7oojtzQE/vKYEEJFjJZ32js2W
```
На эти исправления инструкции я уже направил Pull request.

Так же если не используете проект в маршрутизаторе удалите следующие строки:
```yaml
      # Разрешить маршрутизацию всех IP адресов, в маршрутизатор в любом случае в ручном режиме добавляются маршруты
      # Таким образом можно использовать подключение для полноценного VPN
      - WG_ALLOWED_IPS=0.0.0.0/0
```
Иначе весь трафик пойдет через VPN.

**Reply by Ernieleo** [reply]
Спасибо за помощь, нашёл время и всё легко и просто реализовал.

Для всех кто будет делать это в будущем, думаю стоит добавить в инструкции отдельную сноску о совместимости с приложениями Амнезии: На ПК подключение в приложении AmneziaVPN было успешно выполнено и появилась информация о туннелировании на стороне сервера, но маршрутизация трафика не изменилась, пришлось скачать отдельное приложение под AmneziaWG ([ссылка на Git](https://github.com/amnezia-vpn/amneziawg-windows-client)), на Android такой проблемы не возникло, достаточно было удалить ранее настроенные условия туннелирования внутри приложения и выполнить подключение через конфиг в приложении AmneziaVPN, всё работает отлично.

**Reply by xtrime-ru** [reply]
В приложении AmneziaVPN нужно вручную указать DNS сервер 10.224.0.1 в настройках. Они почему то игнорируют директиву DNS из Wireguard подключения.
